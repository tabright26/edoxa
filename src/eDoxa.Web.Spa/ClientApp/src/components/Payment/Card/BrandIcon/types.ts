import { SizeProp } from "@fortawesome/fontawesome-svg-core";

export type BrandProp = "visa" | "amex" | "mastercard" | "discover";

export interface CardBrandIconProps {
  className: string;
  brand: BrandProp;
  size?: SizeProp;
}
