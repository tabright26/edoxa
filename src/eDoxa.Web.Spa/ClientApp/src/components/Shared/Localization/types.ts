export interface Country {
  readonly name: string;
  readonly twoDigitIso: string;
}

export interface Address {
  country: Country;
  line1: string;
  line2: string | null;
  city: string;
  state: string | null;
  postalCode: string;
}
