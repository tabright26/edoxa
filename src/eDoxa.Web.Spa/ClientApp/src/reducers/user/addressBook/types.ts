import { AxiosActionCreator, AxiosAction } from "interfaces/axios";

export const LOAD_ADDRESS_BOOK = "LOAD_ADDRESS_BOOK";
export const LOAD_ADDRESS_BOOK_SUCCESS = "LOAD_ADDRESS_BOOK_SUCCESS";
export const LOAD_ADDRESS_BOOK_FAIL = "LOAD_ADDRESS_BOOK_FAIL";

export const ADD_ADDRESS = "ADD_ADDRESS";
export const ADD_ADDRESS_SUCCESS = "ADD_ADDRESS_SUCCESS";
export const ADD_ADDRESS_FAIL = "ADD_ADDRESS_FAIL";

export const UPDATE_ADDRESS = "UPDATE_ADDRESS";
export const UPDATE_ADDRESS_SUCCESS = "UPDATE_ADDRESS_SUCCESS";
export const UPDATE_ADDRESS_FAIL = "UPDATE_ADDRESS_FAIL";

export const REMOVE_ADDRESS = "REMOVE_ADDRESS";
export const REMOVE_ADDRESS_SUCCESS = "REMOVE_ADDRESS_SUCCESS";
export const REMOVE_ADDRESS_FAIL = "REMOVE_ADDRESS_FAIL";

type LoadAddressBookType = typeof LOAD_ADDRESS_BOOK | typeof LOAD_ADDRESS_BOOK_SUCCESS | typeof LOAD_ADDRESS_BOOK_FAIL;

interface LoadAddressBookActionCreator extends AxiosActionCreator<LoadAddressBookType> {}

interface LoadAddressBookAction extends AxiosAction<LoadAddressBookType> {}

type AddAddressType = typeof ADD_ADDRESS | typeof ADD_ADDRESS_SUCCESS | typeof ADD_ADDRESS_FAIL;

interface AddAddressActionCreator extends AxiosActionCreator<AddAddressType> {}

interface AddAddressAction extends AxiosAction<AddAddressType> {}

type UpdateAddressType = typeof UPDATE_ADDRESS | typeof UPDATE_ADDRESS_SUCCESS | typeof UPDATE_ADDRESS_FAIL;

interface UpdateAddressActionCreator extends AxiosActionCreator<UpdateAddressType> {}

interface UpdateAddressAction extends AxiosAction<UpdateAddressType> {}

type RemoveAddressType = typeof REMOVE_ADDRESS | typeof REMOVE_ADDRESS_SUCCESS | typeof REMOVE_ADDRESS_FAIL;

interface RemoveAddressActionCreator extends AxiosActionCreator<RemoveAddressType> {}

interface RemoveAddressAction extends AxiosAction<RemoveAddressType> {}

export type AddressBookActionCreators = LoadAddressBookActionCreator | AddAddressActionCreator | UpdateAddressActionCreator | RemoveAddressActionCreator;

export type AddressBookActionTypes = LoadAddressBookAction | AddAddressAction | UpdateAddressAction | RemoveAddressAction;

export interface AddressBookState {}
