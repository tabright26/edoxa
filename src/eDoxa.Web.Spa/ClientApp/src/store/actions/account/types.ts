import { AxiosActionCreator, AxiosAction, AxiosState } from "utils/axios/types";
import { Balance, Bundle, Transaction } from "types";

export type UserAccountBalanceActionCreators =
  | LoadUserMoneyAccountBalanceActionCreator
  | LoadUserTokenAccountBalanceActionCreator;

export const LOAD_USER_MONEY_ACCOUNT_BALANCE =
  "LOAD_USER_MONEY_ACCOUNT_BALANCE";
export const LOAD_USER_MONEY_ACCOUNT_BALANCE_SUCCESS =
  "LOAD_USER_MONEY_ACCOUNT_BALANCE_SUCCESS";
export const LOAD_USER_MONEY_ACCOUNT_BALANCE_FAIL =
  "LOAD_USER_MONEY_ACCOUNT_BALANCE_FAIL";

export type LoadUserMoneyAccountBalanceType =
  | typeof LOAD_USER_MONEY_ACCOUNT_BALANCE
  | typeof LOAD_USER_MONEY_ACCOUNT_BALANCE_SUCCESS
  | typeof LOAD_USER_MONEY_ACCOUNT_BALANCE_FAIL;
export type LoadUserMoneyAccountBalanceActionCreator = AxiosActionCreator<
  LoadUserMoneyAccountBalanceType
>;
export type LoadUserMoneyAccountBalanceAction = AxiosAction<
  LoadUserMoneyAccountBalanceType,
  Balance
>;

export type UserMoneyAccountBalanceActions = LoadUserMoneyAccountBalanceAction;

export const LOAD_USER_TOKEN_ACCOUNT_BALANCE =
  "LOAD_USER_TOKEN_ACCOUNT_BALANCE";
export const LOAD_USER_TOKEN_ACCOUNT_BALANCE_SUCCESS =
  "LOAD_USER_TOKEN_ACCOUNT_BALANCE_SUCCESS";
export const LOAD_USER_TOKEN_ACCOUNT_BALANCE_FAIL =
  "LOAD_USER_TOKEN_ACCOUNT_BALANCE_FAIL";

export type LoadUserTokenAccountBalanceType =
  | typeof LOAD_USER_TOKEN_ACCOUNT_BALANCE
  | typeof LOAD_USER_TOKEN_ACCOUNT_BALANCE_SUCCESS
  | typeof LOAD_USER_TOKEN_ACCOUNT_BALANCE_FAIL;
export type LoadUserTokenAccountBalanceActionCreator = AxiosActionCreator<
  LoadUserTokenAccountBalanceType
>;
export type LoadUserTokenAccountBalanceAction = AxiosAction<
  LoadUserTokenAccountBalanceType,
  Balance
>;

export type UserTokenAccountBalanceActions = LoadUserTokenAccountBalanceAction;

export const USER_ACCOUNT_DEPOSIT_MONEY = "USER_ACCOUNT_DEPOSIT_MONEY";
export const USER_ACCOUNT_DEPOSIT_MONEY_SUCCESS =
  "USER_ACCOUNT_DEPOSIT_MONEY_SUCCESS";
export const USER_ACCOUNT_DEPOSIT_MONEY_FAIL =
  "USER_ACCOUNT_DEPOSIT_MONEY_FAIL";

export const USER_ACCOUNT_DEPOSIT_TOKEN = "USER_ACCOUNT_DEPOSIT_TOKEN";
export const USER_ACCOUNT_DEPOSIT_TOKEN_SUCCESS =
  "USER_ACCOUNT_DEPOSIT_TOKEN_SUCCESS";
export const USER_ACCOUNT_DEPOSIT_TOKEN_FAIL =
  "USER_ACCOUNT_DEPOSIT_TOKEN_FAIL";

export type UserAccountDepositMoneyType =
  | typeof USER_ACCOUNT_DEPOSIT_MONEY
  | typeof USER_ACCOUNT_DEPOSIT_MONEY_SUCCESS
  | typeof USER_ACCOUNT_DEPOSIT_MONEY_FAIL;
export type UserAccountDepositMoneyActionCreator = AxiosActionCreator<
  UserAccountDepositMoneyType
>;
export type UserAccountDepositMoneyAction = AxiosAction<
  UserAccountDepositMoneyType,
  Transaction
>;

export type UserAccountDepositTokenType =
  | typeof USER_ACCOUNT_DEPOSIT_TOKEN
  | typeof USER_ACCOUNT_DEPOSIT_TOKEN_SUCCESS
  | typeof USER_ACCOUNT_DEPOSIT_TOKEN_FAIL;
export type UserAccountDepositTokenActionCreator = AxiosActionCreator<
  UserAccountDepositTokenType
>;
export type UserAccountDepositTokenAction = AxiosAction<
  UserAccountDepositTokenType,
  Transaction
>;

export type UserAccountDepositActionCreators =
  | UserAccountDepositMoneyActionCreator
  | UserAccountDepositTokenActionCreator;
export type UserAccountDepositActions =
  | UserAccountDepositMoneyAction
  | UserAccountDepositTokenAction;

export const LOAD_USER_ACCOUNT_DEPOSIT_MONEY_BUNDLES =
  "LOAD_USER_ACCOUNT_DEPOSIT_MONEY_BUNDLES";
export const LOAD_USER_ACCOUNT_DEPOSIT_MONEY_BUNDLES_SUCCESS =
  "LOAD_USER_ACCOUNT_DEPOSIT_MONEY_BUNDLES_SUCCESS";
export const LOAD_USER_ACCOUNT_DEPOSIT_MONEY_BUNDLES_FAIL =
  "LOAD_USER_ACCOUNT_DEPOSIT_MONEY_BUNDLES_FAIL";

export const LOAD_USER_ACCOUNT_DEPOSIT_TOKEN_BUNDLES =
  "LOAD_USER_ACCOUNT_DEPOSIT_TOKEN_BUNDLES";
export const LOAD_USER_ACCOUNT_DEPOSIT_TOKEN_BUNDLES_SUCCESS =
  "LOAD_USER_ACCOUNT_DEPOSIT_TOKEN_BUNDLES_SUCCESS";
export const LOAD_USER_ACCOUNT_DEPOSIT_TOKEN_BUNDLES_FAIL =
  "LOAD_USER_ACCOUNT_DEPOSIT_TOKEN_BUNDLES_FAIL";

export type LoadUserAccountDepositMoneyBundlesType =
  | typeof LOAD_USER_ACCOUNT_DEPOSIT_MONEY_BUNDLES
  | typeof LOAD_USER_ACCOUNT_DEPOSIT_MONEY_BUNDLES_SUCCESS
  | typeof LOAD_USER_ACCOUNT_DEPOSIT_MONEY_BUNDLES_FAIL;
export type LoadUserAccountDepositMoneyBundlesActionCreator = AxiosActionCreator<
  LoadUserAccountDepositMoneyBundlesType
>;
export type LoadUserAccountDepositMoneyBundlesAction = AxiosAction<
  LoadUserAccountDepositMoneyBundlesType,
  Bundle[]
>;

export type LoadUserAccountDepositTokenBundlesType =
  | typeof LOAD_USER_ACCOUNT_DEPOSIT_TOKEN_BUNDLES
  | typeof LOAD_USER_ACCOUNT_DEPOSIT_TOKEN_BUNDLES_SUCCESS
  | typeof LOAD_USER_ACCOUNT_DEPOSIT_TOKEN_BUNDLES_FAIL;
export type LoadUserAccountDepositTokenBundlesActionCreator = AxiosActionCreator<
  LoadUserAccountDepositTokenBundlesType
>;
export type LoadUserAccountDepositTokenBundlesAction = AxiosAction<
  LoadUserAccountDepositTokenBundlesType,
  Bundle[]
>;

export type UserAccountDepositBundlesActionCreators =
  | LoadUserAccountDepositMoneyBundlesActionCreator
  | LoadUserAccountDepositTokenBundlesActionCreator;
export type UserAccountDepositBundlesActions =
  | LoadUserAccountDepositMoneyBundlesAction
  | LoadUserAccountDepositTokenBundlesAction;
export type UserAccountDepositBundlesState = AxiosState<Bundle[]>;

export const LOAD_USER_ACCOUNT_TRANSACTIONS = "LOAD_USER_ACCOUNT_TRANSACTIONS";
export const LOAD_USER_ACCOUNT_TRANSACTIONS_SUCCESS =
  "LOAD_USER_ACCOUNT_TRANSACTIONS_SUCCESS";
export const LOAD_USER_ACCOUNT_TRANSACTIONS_FAIL =
  "LOAD_USER_ACCOUNT_TRANSACTIONS_FAIL";

export type LoadUserAccountTransactionsType =
  | typeof LOAD_USER_ACCOUNT_TRANSACTIONS
  | typeof LOAD_USER_ACCOUNT_TRANSACTIONS_SUCCESS
  | typeof LOAD_USER_ACCOUNT_TRANSACTIONS_FAIL;
export type LoadUserAccountTransactionsActionCreator = AxiosActionCreator<
  LoadUserAccountTransactionsType
>;
export type LoadUserAccountTransactionsAction = AxiosAction<
  LoadUserAccountTransactionsType,
  Transaction[]
>;

export type UserAccountTransactionsActionCreators = LoadUserAccountTransactionsActionCreator;
export type UserAccountTransactionsActions = LoadUserAccountTransactionsAction;
export type UserAccountTransactionsState = AxiosState<Transaction[]>;

export const USER_ACCOUNT_WITHDRAWAL_MONEY = "USER_ACCOUNT_WITHDRAWAL_MONEY";
export const USER_ACCOUNT_WITHDRAWAL_MONEY_SUCCESS =
  "USER_ACCOUNT_WITHDRAWAL_MONEY_SUCCESS";
export const USER_ACCOUNT_WITHDRAWAL_MONEY_FAIL =
  "USER_ACCOUNT_WITHDRAWAL_MONEY_FAIL";

export type UserAccountWithdrawalMoneyType =
  | typeof USER_ACCOUNT_WITHDRAWAL_MONEY
  | typeof USER_ACCOUNT_WITHDRAWAL_MONEY_SUCCESS
  | typeof USER_ACCOUNT_WITHDRAWAL_MONEY_FAIL;
export type UserAccountWithdrawalMoneyActionCreator = AxiosActionCreator<
  UserAccountWithdrawalMoneyType
>;
export type UserAccountWithdrawalMoneyAction = AxiosAction<
  UserAccountWithdrawalMoneyType,
  Transaction
>;

export type UserAccountWithdrawalActionCreators = UserAccountWithdrawalMoneyActionCreator;
export type UserAccountWithdrawalActions = UserAccountWithdrawalMoneyAction;

export const LOAD_USER_ACCOUNT_WITHDRAWAL_MONEY_BUNDLES =
  "LOAD_USER_ACCOUNT_WITHDRAWAL_MONEY_BUNDLES";
export const LOAD_USER_ACCOUNT_WITHDRAWAL_MONEY_BUNDLES_SUCCESS =
  "LOAD_USER_ACCOUNT_WITHDRAWAL_MONEY_BUNDLES_SUCCESS";
export const LOAD_USER_ACCOUNT_WITHDRAWAL_MONEY_BUNDLES_FAIL =
  "LOAD_USER_ACCOUNT_WITHDRAWAL_MONEY_BUNDLES_FAIL";

export type LoadUserAccountWithdrawalMoneyBundlesType =
  | typeof LOAD_USER_ACCOUNT_WITHDRAWAL_MONEY_BUNDLES
  | typeof LOAD_USER_ACCOUNT_WITHDRAWAL_MONEY_BUNDLES_SUCCESS
  | typeof LOAD_USER_ACCOUNT_WITHDRAWAL_MONEY_BUNDLES_FAIL;
export type LoadUserAccountWithdrawalMoneyBundlesActionCreator = AxiosActionCreator<
  LoadUserAccountWithdrawalMoneyBundlesType
>;
export type LoadUserAccountWithdrawalMoneyBundlesAction = AxiosAction<
  LoadUserAccountWithdrawalMoneyBundlesType,
  Bundle[]
>;

export type UserAccountWithdrawalBundlesActionCreators = LoadUserAccountWithdrawalMoneyBundlesActionCreator;
export type UserAccountWithdrawalBundlesActions = LoadUserAccountWithdrawalMoneyBundlesAction;
export type UserAccountWithdrawalBundlesState = AxiosState<Bundle[]>;
