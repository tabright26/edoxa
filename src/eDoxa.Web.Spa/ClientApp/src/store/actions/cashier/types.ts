import { AxiosActionCreator, AxiosAction } from "utils/axios/types";
import { Transaction, Promotion } from "types";

export const DEPOSIT_TRANSACTION = "DEPOSIT_TRANSACTION";
export const DEPOSIT_TRANSACTION_SUCCESS = "DEPOSIT_TRANSACTION_SUCCESS";
export const DEPOSIT_TRANSACTION_FAIL = "DEPOSIT_TRANSACTION_FAIL";

export type DepositTransactionType =
  | typeof DEPOSIT_TRANSACTION
  | typeof DEPOSIT_TRANSACTION_SUCCESS
  | typeof DEPOSIT_TRANSACTION_FAIL;
export type DepositTransactionActionCreator = AxiosActionCreator<
  DepositTransactionType
>;
export type DepositTransactionAction = AxiosAction<DepositTransactionType>;

export const WITHDRAW_TRANSACTION = "WITHDRAW_TRANSACTION";
export const WITHDRAW_TRANSACTION_SUCCESS = "WITHDRAW_TRANSACTION_SUCCESS";
export const WITHDRAW_TRANSACTION_FAIL = "WITHDRAW_TRANSACTION_FAIL";

export type WithdrawTransactionType =
  | typeof WITHDRAW_TRANSACTION
  | typeof WITHDRAW_TRANSACTION_SUCCESS
  | typeof WITHDRAW_TRANSACTION_FAIL;
export type WithdrawTransactionActionCreator = AxiosActionCreator<
  WithdrawTransactionType
>;
export type WithdrawTransactionAction = AxiosAction<WithdrawTransactionType>;

export const LOAD_USER_TRANSACTION_HISTORY = "LOAD_USER_TRANSACTION_HISTORY";
export const LOAD_USER_TRANSACTION_HISTORY_SUCCESS =
  "LOAD_USER_TRANSACTION_HISTORY_SUCCESS";
export const LOAD_USER_TRANSACTION_HISTORY_FAIL =
  "LOAD_USER_TRANSACTION_HISTORY_FAIL";

export type LoadUserTransactionHistoryType =
  | typeof LOAD_USER_TRANSACTION_HISTORY
  | typeof LOAD_USER_TRANSACTION_HISTORY_SUCCESS
  | typeof LOAD_USER_TRANSACTION_HISTORY_FAIL;
export type LoadUserTransactionHistoryActionCreator = AxiosActionCreator<
  LoadUserTransactionHistoryType
>;
export type LoadUserTransactionHistoryAction = AxiosAction<
  LoadUserTransactionHistoryType,
  Transaction[]
>;

export const REDEEM_PROMOTION = "REDEEM_PROMOTION";
export const REDEEM_PROMOTION_SUCCESS = "REDEEM_PROMOTION_SUCCESS";
export const REDEEM_PROMOTION_FAIL = "REDEEM_PROMOTION_FAIL";

export type RedeemPromotionType =
  | typeof REDEEM_PROMOTION
  | typeof REDEEM_PROMOTION_SUCCESS
  | typeof REDEEM_PROMOTION_FAIL;
export type RedeemPromotionActionCreator = AxiosActionCreator<
  RedeemPromotionType
>;
export type RedeemPromotionAction = AxiosAction<RedeemPromotionType, Promotion>;
