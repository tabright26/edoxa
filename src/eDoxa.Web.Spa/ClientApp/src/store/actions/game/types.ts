import { AxiosActionCreator, AxiosAction } from "utils/axios/types";

export const GENERATE_GAME_AUTHENTICATION = "GENERATE_GAME_AUTHENTICATION";
export const GENERATE_GAME_AUTHENTICATION_SUCCESS =
  "GENERATE_GAME_AUTHENTICATION_SUCCESS";
export const GENERATE_GAME_AUTHENTICATION_FAIL =
  "GENERATE_GAME_AUTHENTICATION_FAIL";

export const VALIDATE_GAME_AUTHENTICATION = "VALIDATE_GAME_AUTHENTICATION";
export const VALIDATE_GAME_AUTHENTICATION_SUCCESS =
  "VALIDATE_GAME_AUTHENTICATION_SUCCESS";
export const VALIDATE_GAME_AUTHENTICATION_FAIL =
  "VALIDATE_GAME_AUTHENTICATION_FAIL";

export type GenerateGameAuthenticationType =
  | typeof GENERATE_GAME_AUTHENTICATION
  | typeof GENERATE_GAME_AUTHENTICATION_SUCCESS
  | typeof GENERATE_GAME_AUTHENTICATION_FAIL;
export type GenerateGameAuthenticationActionCreator = AxiosActionCreator<
  GenerateGameAuthenticationType
>;
export type GenerateGameAuthenticationAction = AxiosAction<
  GenerateGameAuthenticationType
>;

export type ValidateGameAuthenticationType =
  | typeof VALIDATE_GAME_AUTHENTICATION
  | typeof VALIDATE_GAME_AUTHENTICATION_SUCCESS
  | typeof VALIDATE_GAME_AUTHENTICATION_FAIL;
export type ValidateGameAuthenticationActionCreator = AxiosActionCreator<
  ValidateGameAuthenticationType
>;
export type ValidateGameAuthenticationAction = AxiosAction<
  ValidateGameAuthenticationType
>;

export type GameAuthenticationActionCreators =
  | ValidateGameAuthenticationActionCreator
  | GenerateGameAuthenticationActionCreator;
export type GameAuthenticationActions =
  | ValidateGameAuthenticationAction
  | GenerateGameAuthenticationAction;

export const UNLINK_GAME_CREDENTIAL = "UNLINK_GAME_CREDENTIAL";
export const UNLINK_GAME_CREDENTIAL_SUCCESS = "UNLINK_GAME_CREDENTIAL_SUCCESS";
export const UNLINK_GAME_CREDENTIAL_FAIL = "UNLINK_GAME_CREDENTIAL_FAIL";

export type UnlinkGameCredentialType =
  | typeof UNLINK_GAME_CREDENTIAL
  | typeof UNLINK_GAME_CREDENTIAL_SUCCESS
  | typeof UNLINK_GAME_CREDENTIAL_FAIL;
export type UnlinkGameCredentialActionCreator = AxiosActionCreator<
  UnlinkGameCredentialType
>;
export type UnlinkGameCredentialAction = AxiosAction<UnlinkGameCredentialType>;

export type GameCredentialActionCreators = UnlinkGameCredentialActionCreator;
export type GameCredentialActions = UnlinkGameCredentialAction;
