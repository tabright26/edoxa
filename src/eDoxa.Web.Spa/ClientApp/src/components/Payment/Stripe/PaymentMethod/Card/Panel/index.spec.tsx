import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import Cards from ".";
import { StripePaymentMethodsState } from "store/root/payment/stripe/paymentMethod/types";

it("renders without crashing", () => {
  //Arrange
  const paymentMethods: StripePaymentMethodsState = {
    data: [
      {
        id: "0",
        card: {
          brand: "visa",
          country: "CA",
          expMonth: 6,
          expYear: 2015,
          last4: "4242"
        }
      }
    ],
    loading: false,
    error: null
  };

  const store: any = {
    getState: () => {
      return {
        root: {
          payment: {
            stripe: {
              paymentMethods
            }
          }
        }
      };
    },
    dispatch: action => {},
    subscribe: () => {}
  };

  //Act
  const tree = renderer
    .create(
      <Provider store={store}>
        <Cards />
      </Provider>
    )
    .toJSON();

  //Assert
  expect(tree).toMatchSnapshot();
});
