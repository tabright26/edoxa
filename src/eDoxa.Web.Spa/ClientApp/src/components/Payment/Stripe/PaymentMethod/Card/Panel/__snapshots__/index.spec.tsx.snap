// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="card-accent-primary undefined card"
>
  <div
    className="d-flex card-header"
  >
    <strong
      className="text-uppercase my-auto"
    >
      CREDIT CARDS
    </strong>
    <small
      className="ml-2 my-auto text-muted"
    >
      (
      1
      /
      5
      )
    </small>
    <button
      aria-label={null}
      className="p-0 ml-auto my-auto btn btn-link btn-sm"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <small
        className="text-uppercase"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-plus fa-w-14 "
          data-icon="plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         
        ADD A NEW CREDIT CARD
      </small>
    </button>
  </div>
  <div
    className="card-body"
  >
    <dl
      className="row mb-0"
    >
      <dd
        className="col-sm-4 m-0 d-flex"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-cc-visa fa-w-18 fa-2x my-auto"
          data-icon="cc-visa"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M470.1 231.3s7.6 37.2 9.3 45H446c3.3-8.9 16-43.5 16-43.5-.2.3 3.3-9.1 5.3-14.9l2.8 13.4zM576 80v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V80c0-26.5 21.5-48 48-48h480c26.5 0 48 21.5 48 48zM152.5 331.2L215.7 176h-42.5l-39.3 106-4.3-21.5-14-71.4c-2.3-9.9-9.4-12.7-18.2-13.1H32.7l-.7 3.1c15.8 4 29.9 9.8 42.2 17.1l35.8 135h42.5zm94.4.2L272.1 176h-40.2l-25.1 155.4h40.1zm139.9-50.8c.2-17.7-10.6-31.2-33.7-42.3-14.1-7.1-22.7-11.9-22.7-19.2.2-6.6 7.3-13.4 23.1-13.4 13.1-.3 22.7 2.8 29.9 5.9l3.6 1.7 5.5-33.6c-7.9-3.1-20.5-6.6-36-6.6-39.7 0-67.6 21.2-67.8 51.4-.3 22.3 20 34.7 35.2 42.2 15.5 7.6 20.8 12.6 20.8 19.3-.2 10.4-12.6 15.2-24.1 15.2-16 0-24.6-2.5-37.7-8.3l-5.3-2.5-5.6 34.9c9.4 4.3 26.8 8.1 44.8 8.3 42.2.1 69.7-20.8 70-53zM528 331.4L495.6 176h-31.1c-9.6 0-16.9 2.8-21 12.9l-59.7 142.5H426s6.9-19.2 8.4-23.3H486c1.2 5.5 4.8 23.3 4.8 23.3H528z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <span
          className="ml-2 my-auto"
        >
          XXXX XXXX XXXX 4242
        </span>
      </dd>
      <dd
        className="col-sm-2 m-0 d-flex"
      >
        <span
          className="my-auto mr-2 text-muted"
        >
          Expiration:
        </span>
        <span
          className="my-auto"
        >
          06
        </span>
        <span
          className="my-auto"
        >
          /
        </span>
        <span
          className="my-auto"
        >
          15
        </span>
      </dd>
      <dd
        className="col-sm-6 mb-0 d-flex"
      >
        <button
          aria-label={null}
          className="p-0 ml-auto my-auto btn btn-link btn-sm"
          disabled={null}
          onClick={[Function]}
          type="button"
        >
          <small
            className="text-uppercase"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-times fa-w-11 "
              data-icon="times"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 352 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             
            REMOVE
          </small>
        </button>
        <button
          aria-label={null}
          className="p-0 ml-auto my-auto btn btn-link btn-sm"
          disabled={null}
          onClick={[Function]}
          type="button"
        >
          <small
            className="text-uppercase"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-edit fa-w-18 "
              data-icon="edit"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             
            UPDATE
          </small>
        </button>
      </dd>
    </dl>
  </div>
</div>
`;
